tabular/src/autogluon/tabular/models/rf/rf_model.py

```diff
@@ -18,6 +19,121 @@
 logger = logging.getLogger(__name__)
 
 
+# FIXME: DOESNT WORK ON REGRESSION PREDICTIONS
+class RFOnnxPredictor:
+    def __init__(self, obj, model):
+        self.num_classes = obj.num_classes
+        self.model = model
+
+    def predict(self, X):
+        input_name = self.model.get_inputs()[0].name
+        label_name = self.model.get_outputs()[0].name
+        return self.model.run([label_name], {input_name: X})[0]
+
+    def predict_proba(self, X):
+        input_name = self.model.get_inputs()[0].name
+        label_name = self.model.get_outputs()[1].name
+        pred_proba = self.model.run([label_name], {input_name: X})[0]
+        pred_proba = np.array([[r[i] for i in range(self.num_classes)] for r in pred_proba])
+        return pred_proba
+
+
+class RFNativeCompiler:
+    name = 'native'
+    save_in_pkl = True
+
+    @staticmethod
+    def can_compile():
+        return True
+
+    @staticmethod
+    def compile(obj, path: str):
+        from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+        from sklearn.ensemble import ExtraTreesRegressor, ExtraTreesClassifier
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+        if isinstance(obj.model, (RandomForestClassifier, RandomForestRegressor,
+                                  ExtraTreesClassifier, ExtraTreesRegressor)):
+            with open(path + 'model_native.pkl', 'wb') as fp:
+                fp.write(pickle.dumps(obj))
+        else:
+            pkl = None
+            with open(path + 'model_native.pkl', 'rb') as fp:
+                pkl = fp.read()
+            return pickle.loads(pkl).model
+        return obj.model
+
+    @staticmethod
+    def load(obj, path: str):
+        pkl = None
+        with open(path + 'model_native.pkl', 'rb') as fp:
+            pkl = fp.read()
+        model = pickle.loads(pkl)
+        return model.model
+
+
+class InferenceSessionWrapper:
+    """
+    Wrap around InferenceSession in onnxruntime, since it cannot be pickled.
+    See https://github.com/microsoft/onnxruntime/issues/10097
+    """
+    def __init__(self, onnx_bytes):
+        import onnxruntime as rt
+        self.sess = rt.InferenceSession(onnx_bytes.SerializeToString())
+        self.onnx_bytes = onnx_bytes
```

Comment:
Is this doubling our memory usage by keeping this in memory after creating the session? If it is, we probably don't want to store the variable.

Reply:
I tested in my local env that we actually don't need to duplicate the onnx parameters in the inference session class. This extra copy has been removed.